ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -227,7 +228,7 @@ class SimpleOperationTracker implements OperationTracker {
     // Order the replicas so that local healthy replicas are ordered and returned first,
     // then the remote healthy ones, and finally the possibly down ones.
     List<? extends ReplicaId> replicas =
-        routerConfig.routerGetEligibleReplicasByStateEnabled ? eligibleReplicas : allReplicas;
+        routerConfig.routerGetEligibleReplicasByStateEnabled ? eligibleReplicas : partitionId.getReplicaIds();
```

Comment:
What's the reason to change allReplicas to getReplicaIds.
Before looks like we want to have local healthy replicas at the beginning. After the change, the order will be different. Is that a concern?

Reply:
the order will be enforced the the code after this line.

The reason for this change is for test. In some tests, we set up error responses in certain order that is different from the allReplicas, but same with partitionId.getReplicaIds().

Reply:
What about the following code? Since the replicas list's order is changed. the order we put to the Pool is different. Is this a concern?
        if (isLocalDcReplica) {
          addToBeginningOfPool(replicaId);
        } 


    for (ReplicaId replicaId : replicas) {
      examinedReplicas.add(replicaId);
      String replicaDcName = replicaId.getDataNodeId().getDatacenterName();
      boolean isLocalDcReplica = replicaDcName.equals(datacenterName);
      boolean isOriginatingDcReplica = replicaDcName.equals(originatingDcName);
      if (isOriginatingDcReplica) {
        numReplicasInOriginatingDc++;
      }
      if (!replicaId.isDown()) {
        if (isLocalDcReplica) {
          addToBeginningOfPool(replicaId);
        } else if (crossColoEnabled && isOriginatingDcReplica) {
          addToEndOfPool(replicaId);
        } else if (crossColoEnabled) {
          backupReplicas.addFirst(replicaId);
        }
      } else {
        if (isLocalDcReplica) {
          downReplicas.addFirst(replicaId);
        } else if (crossColoEnabled) {
          downReplicas.addLast(replicaId);
        }
      }
    }

Reply:
and the allReplicas doesn't have a deterministic order anyway. It has replicas from all state, leader, standby, bootstrap, inactive, offline. When allReplicas is created, it's created by iterating through a unordered map, which doesn't guarantee the order of each state.